vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/MessageInformationDataFactory.kt

```diff
@@ -72,8 +74,14 @@ class MessageInformationDataFactory @Inject constructor(
                 prevDisplayableEvent?.root?.localDateTime()?.toLocalDate() != date.toLocalDate()
 
         val time = dateFormatter.format(event.root.originServerTs, DateFormatKind.MESSAGE_SIMPLE)
-        val e2eDecoration = getE2EDecoration(roomSummary, event)
-
+        val e2eDecoration = getE2EDecoration(roomSummary, lastEdit ?: event.root)
+        val senderId = if (event.isEncrypted()) {
```

Comment:
Maybe we should extract this inside a dedicated private method such as `getSenderId()` to reduce complexity of the `create` method?

Reply:
done